src/storage/engine/mod.rs

```diff
@@ -369,8 +381,23 @@ mod tests {
         assert_near_seek(cursor_mut, b"y", (b"z", b"2"));
         assert_near_seek(cursor_mut, b"x\x00", (b"z", b"2"));
         assert!(!cursor_mut.near_seek(&make_key(b"z\x00")).unwrap());
+        // Insert many key-values between 'x' and 'z' then near_seek will fallback to seek.
```

Comment:
I think that would be very strange. In addition, tests could run in multiple threads, we will not able to confirm it is increased by this test.